core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/commandqueue/handlers/CancelTransferCommandHandler.java

```diff
@@ -0,0 +1,22 @@
+package org.eclipse.dataspaceconnector.transfer.core.commandqueue.handlers;
+
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcess;
+import org.eclipse.dataspaceconnector.transfer.core.commandqueue.commands.CancelTransferCommand;
+
+public class CancelTransferCommandHandler extends TransferProcessCommandHandler<CancelTransferCommand> {
+
+    public CancelTransferCommandHandler(TransferProcessStore store) {
+        super(store);
+    }
+
+    @Override
+    public Class<CancelTransferCommand> getType() {
+        return CancelTransferCommand.class;
+    }
+
+    @Override
+    protected void modify(TransferProcess process) {
+        process.transitionError("Cancelled");
```

Comment:
Should the TP be transitioned to an error state or should we have a canceled state?

Reply:
that would certainly be more appropriate. Will do in #515